db/hints/sync_point.cc

```diff
@@ -41,6 +41,8 @@ namespace hints {
 //       Flattened representation was chosen in order to save space on
```

Comment:
The commit message in `db: hints: add the version_size constant` refers to some things that do not exist at the parent of this commit.

> As the V2 format of encoding sync points introduces the new checksum_size
constant, we can additionally add the version_size constant to increase
consistency and readability.

What's the "V2 format"? What is "the new checksum_size"?

Reply:
I changed this commit message. It should be clear now that it refers to the changes added in the 2 next commits.

Reply:
I don't agree. I think doing small refactors/cleanups first and then introducing functionality results in cleaner, more understandable diffs.

On one hand, I think the cover letter (PR description) should mention that the PR introduces the V2 format which adds a checksum, so that reviewers are surprised less when they see mentions of V2.

On the other hand, perhaps this commit message is too specific. Personally, I would avoid mentioning the `checksum_size` field, just that we want to extend the format and add one more field, so we do a cleanup of the existing field first.